crates/router/src/core/payments/flows/setup_mandate_flow.rs

```diff
@@ -86,14 +87,76 @@ impl Feature<api::SetupMandate, types::SetupMandateRequestData> for types::Setup
         ))
         .await?;
 
-        mandate::mandate_procedure(
-            state,
-            resp,
-            maybe_customer,
-            pm_id,
-            connector.merchant_connector_id.clone(),
-        )
-        .await
+        if let Some(mandate_id) = self
+            .request
+            .setup_mandate_details
+            .as_ref()
+            .and_then(|mandate_data| mandate_data.update_mandate_id.clone())
+        {
+            let mandate = state
+                .store
+                .find_mandate_by_merchant_id_mandate_id(&merchant_account.merchant_id, &mandate_id)
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::MandateNotFound)?;
+
+            let profile_id = mandate::helpers::get_profile_id_for_mandate(
+                state,
+                merchant_account,
+                mandate.clone(),
+            )
+            .await?;
+            let connector_integration: services::BoxedConnectorIntegration<
+                '_,
+                types::api::MandateRevoke,
+                types::MandateRevokeRequestData,
+                types::MandateRevokeResponseData,
+            > = connector.connector.get_connector_integration();
+            let merchant_connector_account = helpers::get_merchant_connector_account(
+                state,
+                &merchant_account.merchant_id,
+                None,
+                key_store,
+                &profile_id,
+                &mandate.connector,
+                mandate.merchant_connector_id.as_ref(),
```

Comment:
move this logic to a separate function

Reply:
Will pick this up in this [issue](https://github.com/juspay/hyperswitch/issues/3464)